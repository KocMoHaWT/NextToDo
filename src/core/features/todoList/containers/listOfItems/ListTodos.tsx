import React from 'react';
import { ToDo } from '../../../../shared/interfaces/toDo';
import { TodoItem } from '../listItem/TodoItem';

interface ListTodosProps {
  todos: ToDo[];
  onDelete?: (id: number) => void;
  onComplete?: (id: number) => void;
}
export const ListTodos: React.FunctionComponent<ListTodosProps> = ({ onDelete, onComplete, todos }) => {
  return (
    <ul>
      {todos.map((todo: ToDo) => (
        <TodoItem
          todo={todo}
          onDelete={(): void => onDelete(todo.id)}
          onComplete={(): void => onComplete(todo.id)}
          key={todo.id + todo.info}
        />
      ))}
    </ul>
  );
};
