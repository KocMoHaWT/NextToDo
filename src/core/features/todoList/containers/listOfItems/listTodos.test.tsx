import { cleanup } from '@testing-library/react';
import React from 'react';
import { create } from 'react-test-renderer';
import { ListTodos } from './ListTodos';

const testLabel = 'djf;lakjdf';

const todoList = [
  { id: 0, info: testLabel, completed: false },
  { id: 1, info: testLabel, completed: false },
  { id: 2, info: testLabel, completed: true },
];

describe('Button', () => {
  afterEach(cleanup);

  it('list of items renders correctly', () => {
    const tree = create(<ListTodos todos={todoList} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
