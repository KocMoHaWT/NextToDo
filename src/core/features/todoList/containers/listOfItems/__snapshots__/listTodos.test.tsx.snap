// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button list of items renders correctly 1`] = `
<ul>
  <li
    className="listTodo"
  >
    djf;lakjdf
    <div
      className="buttons"
    >
      <label
        className="container container"
      >
        <span
          className="label"
        >
          
        </span>
        <input
          className="checkBox"
          defaultChecked={false}
          name="complete todo"
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="checkmark"
        />
      </label>
      <button
        className="button unborder deleteBtn"
        onClick={[Function]}
        type="button"
      >
        ×
      </button>
    </div>
  </li>
  <li
    className="listTodo"
  >
    djf;lakjdf
    <div
      className="buttons"
    >
      <label
        className="container container"
      >
        <span
          className="label"
        >
          
        </span>
        <input
          className="checkBox"
          defaultChecked={false}
          name="complete todo"
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="checkmark"
        />
      </label>
      <button
        className="button unborder deleteBtn"
        onClick={[Function]}
        type="button"
      >
        ×
      </button>
    </div>
  </li>
  <li
    className="listTodo completed"
  >
    djf;lakjdf
    <div
      className="buttons"
    >
      <label
        className="container container"
      >
        <span
          className="label"
        >
          
        </span>
        <input
          className="checkBox"
          defaultChecked={true}
          name="complete todo"
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="checkmark"
        />
      </label>
      <button
        className="button unborder deleteBtn"
        onClick={[Function]}
        type="button"
      >
        ×
      </button>
    </div>
  </li>
</ul>
`;

exports[`Button todo item renders correctly 1`] = `
<ul>
  <li
    className="listTodo"
  >
    djf;lakjdf
    <div
      className="buttons"
    >
      <label
        className="container container"
      >
        <span
          className="label"
        >
          
        </span>
        <input
          className="checkBox"
          defaultChecked={false}
          name="complete todo"
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="checkmark"
        />
      </label>
      <button
        className="button unborder deleteBtn"
        onClick={[Function]}
        type="button"
      >
        ×
      </button>
    </div>
  </li>
  <li
    className="listTodo"
  >
    djf;lakjdf
    <div
      className="buttons"
    >
      <label
        className="container container"
      >
        <span
          className="label"
        >
          
        </span>
        <input
          className="checkBox"
          defaultChecked={false}
          name="complete todo"
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="checkmark"
        />
      </label>
      <button
        className="button unborder deleteBtn"
        onClick={[Function]}
        type="button"
      >
        ×
      </button>
    </div>
  </li>
  <li
    className="listTodo completed"
  >
    djf;lakjdf
    <div
      className="buttons"
    >
      <label
        className="container container"
      >
        <span
          className="label"
        >
          
        </span>
        <input
          className="checkBox"
          defaultChecked={true}
          name="complete todo"
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="checkmark"
        />
      </label>
      <button
        className="button unborder deleteBtn"
        onClick={[Function]}
        type="button"
      >
        ×
      </button>
    </div>
  </li>
</ul>
`;
